internal/msalbase/Account.go

```diff
@@ -38,56 +38,21 @@ func NewAccount(homeAccountID, env, realm, localAccountID, authorityType, userna
 }
 
 // CreateKey creates the key for storing accounts in the cache.
-func (acc *Account) CreateKey() string {
+func (acc Account) CreateKey() string {
 	return strings.Join([]string{acc.HomeAccountID, acc.Environment, acc.Realm}, CacheKeySeparator)
 }
 
 //GetUsername returns the username of an account
-func (acc *Account) GetUsername() string {
+func (acc Account) GetUsername() string {
 	return acc.PreferredUsername
 }
 
 //GetHomeAccountID returns the home account ID of an account
-func (acc *Account) GetHomeAccountID() string {
+func (acc Account) GetHomeAccountID() string {
 	return acc.HomeAccountID
 }
 
 //GetEnvironment returns the environment of an account
-func (acc *Account) GetEnvironment() string {
+func (acc Account) GetEnvironment() string {
 	return acc.Environment
 }
-
-//PopulateFromJSONMap populates an account object from a map (used for cache deserialization)
-func (acc *Account) PopulateFromJSONMap(j map[string]interface{}) error {
```

Comment:
Do you have an idea of how the perf changes ? I think that the original method does not use reflection.

Reply:
It should be faster on encode/decode (though I haven't benchmarked it, but should be a slight edge).   So this was a double conversion, once reading everything into a type and then stuffing it into an interface{}, which did a heap allocation for all fields, regardless.  Assertions are reflections of a type underneath too.  

That speed will probably be irrelevant.  Bigger problems would be the creation of so many objects that have to get gc'd.  Lots of small objects on the gc is the Go performance killer.

I've actually got a few more ideas on how to reduce allocations from this with AdditionalFields, but I'll add that "upgrade" later.

I can't say for this service, but as far as raw speed goes, I found in Kusto that the network couldn't send fast enough to make a difference in speed on any of these conversion methods, it was the biggest limiter.  However, allocations are where you can affect overall performance.  

Here's a question for some future changes, does MSAL backends support gzip or deflate compression?  If so I should add that in for client sends (I'm going to add it for returns, but that gets negotiated where the send does not).  A fix I put in on Kusto's backend gave us 2x performance with compression on their apps and web tooling.

Reply:
Thanks @element-of-surprise . CCing @pmaytak who's looking at perf in MSAL .NET

We may want to invest into a similar approach for the other MSAL libraries. I fear however this is pretty complex, so I'm trying to better understand the return on investment here (outside maintainability). 

I do not know if gzip / deflate is supported by AAD services.